lfs/client.go

```diff
@@ -422,15 +431,15 @@ func doApiRequestWithRedirects(req *http.Request, creds Creds, via []*http.Reque
 			return res, Errorf(err, err.Error())
 		}
 
-		return doApiRequestWithRedirects(redirectedReq, redirectedCreds, via)
+		return doApiRequestWithRedirects(redirectedReq, via, useCreds)
 	}
 
 	return res, nil
 }
 
-func doApiRequest(req *http.Request, creds Creds) (*http.Response, *objectResource, *WrappedError) {
+func doApiRequest(req *http.Request) (*http.Response, *objectResource, *WrappedError) {
```

Comment:
These and the batch api calls will be unified very soon, putting everything through the batch api. I wouldn't spend any time trying to build an abstraction that covers both cases.